src/compiler/sourcemapDecoder.ts

```diff
@@ -61,7 +61,7 @@ namespace ts.sourcemaps {
 
     export function decode(host: SourceMapDecodeHost, mapPath: string, map: SourceMapData, program?: Program, fallbackCache = createSourceFileLikeCache(host)): SourceMapper {
         const currentDirectory = getDirectoryPath(mapPath);
-        const sourceRoot = map.sourceRoot || currentDirectory;
+        const sourceRoot = map.sourceRoot ? toPath(map.sourceRoot, currentDirectory, p => host.getCanonicalFileName(p)) : currentDirectory;

```

Comment:
Within this file, `sourceRoot` is passed into `comparePaths`, and as a base for `toPath`, so in all places it would just be canonicalized later anyway. By canonicalizing it early, hopefully work is repeated less.